ta4j-core/src/main/java/org/ta4j/core/num/DoubleNum.java

```diff
@@ -23,47 +23,73 @@
  */
 package org.ta4j.core.num;
 
-import static org.ta4j.core.num.NaN.NaN;
-
+import java.util.Map;
+import java.util.Objects;
+import java.util.WeakHashMap;
 import java.util.function.Function;
 
+import static org.ta4j.core.num.NaN.NaN;
+
 /**
  * Representation of Double. High performance, lower precision.
  *
  * @apiNote the delegate should never become a NaN value. No self NaN checks
  *          provided
  */
 public class DoubleNum implements Num {
+    private static final DoubleNum DZERO = new DoubleNum(0d);
```

Comment:
Please don't prefix those properties by "D". It makes it not very readable. Simply use ZERO, ONE, NEGATIVE_ZERO, HUNDRED, etc.